Extensions/ValuesOfMultipleObjects.json

```
@@ -0,0 +1,1796 @@
+  "extensionNamespace": "",
+  "fullName": "Values of multiple objects",
+  "helpPath": "",
+  "iconUrl": "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0ibWRpLWRvdHMtaGV4YWdvbiIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiPjxwYXRoIGQ9Ik0xNiAxMkMxNiAxMC45IDE2LjkgMTAgMTggMTBTMjAgMTAuOSAyMCAxMiAxOS4xIDE0IDE4IDE0IDE2IDEzLjEgMTYgMTJNMTAgMTJDMTAgMTAuOSAxMC45IDEwIDEyIDEwUzE0IDEwLjkgMTQgMTIgMTMuMSAxNCAxMiAxNCAxMCAxMy4xIDEwIDEyTTQgMTJDNCAxMC45IDQuOSAxMCA2IDEwUzggMTAuOSA4IDEyIDcuMSAxNCA2IDE0IDQgMTMuMSA0IDEyTTEzIDE4QzEzIDE2LjkgMTMuOSAxNiAxNSAxNlMxNyAxNi45IDE3IDE4IDE2LjEgMjAgMTUgMjAgMTMgMTkuMSAxMyAxOE03IDE4QzcgMTYuOSA3LjkgMTYgOSAxNlMxMSAxNi45IDExIDE4IDEwLjEgMjAgOSAyMCA3IDE5LjEgNyAxOE0xMyA2QzEzIDQuOSAxMy45IDQgMTUgNFMxNyA0LjkgMTcgNiAxNi4xIDggMTUgOCAxMyA3LjEgMTMgNk03IDZDNyA0LjkgNy45IDQgOSA0UzExIDQuOSAxMSA2IDEwLjEgOCA5IDggNyA3LjEgNyA2IiAvPjwvc3ZnPg==",
+  "name": "ValuesOfMultipleObjects",
+  "previewIconUrl": "https://resources.gdevelop-app.com/assets/Icons/dots-hexagon.svg",
+  "shortDescription": "Values of picked object instances (including position and velocity)",
+  "version": "1.1.0",
+  "tags": [
+    "objects",
+    "expression",
+    "calculate",
+    "center",
+    "middle",
+    "average",
+    "position",
+    "flocking",
+    "camera",
+    "mean",
+    "median",
+    "max",
+    "min",
+    "velocity"
+  ],
+  "authorIds": [
+    "gqDaZjCfevOOxBYkK6zlhtZnXCg1",
+    "2OwwM8ToR9dx9RJ2sAKTcrLmCB92"
+  ],
+  "dependencies": [],
+  "eventsFunctions": [
+    {
+      "description": "Minimum X position of picked object instances (using AABB of objects)",
+      "fullName": "Minimum X position of picked object instances (using AABB of objects)",
+      "functionType": "Expression",
+      "name": "MinX_AABB",
+      "private": false,
+      "sentence": "",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "// Get the first instance picked\nconst object = objects[0];\n\n// If the instance doesn't exist we exit this function, returnValue will be 0\nif (!object) return;\n\n// Get the AABB of the object\nconst aabb = object.getAABB();\n\n// Get the layer of the current object\nconst layer = runtimeScene.getLayer(object.getLayer());\n\neventsFunctionContext.returnValue = aabb.min[0];\n",
```

The Javascript code gets the layer and does nothing with it, I'd remove that line (same on other expressions using this code), alongside unnecessary comments (commenting self-explanatory code, like "This gets the AABB of the object" for `object.getAABB()` just bloats the extension, the end game, and makes it take longer to read the JavaScript code as you take the time to read the comment alongside the code where the comment doesn't provide value towards the understanding of the code)
```suggestion
          "inlineCode": "// Get the currently picked instance\nconst object = objects[0];\nif (!object) return;\n\nconst aabb = object.getAABB();\n\neventsFunctionContext.returnValue = aabb.min[0];\n",
```

---

Good catch on the layer function, it is not used so I removed those lines from all functions.

I actually like how Bouh commented the rest of that function.